vividus-plugin-browserstack/src/main/java/org/vividus/selenium/browserstack/BrowserStackLocalManager.java

```diff
@@ -0,0 +1,195 @@
+    }
+
+    @Override
+    public void stop()
+    {
+        if (isStarted())
+        {
+            synchronized (activeConnections)
+            {
+                if (isStarted())
+                {
+                    BrowserStackLocalConnection connection = getCurrentConnection();
+                    if (connection.decreaseSessionCount() == 0)
+                    {
+                        try
+                        {
+                            connection.stopConnection();
+                        }
+                        catch (Exception e)
+                        {
+                            throw new BrowserStackLocalException(e);
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    @Override
+    public boolean isStarted()
+    {
+        return activeConnections.containsValue(getCurrentConnection());
+    }
+
+    private BrowserStackLocalConnection getCurrentConnection()
+    {
+        return testContext.get(KEY);
+    }
+
+    private void putCurrentConnection(BrowserStackLocalConnection connection)
+    {
+        connection.increaseSessionCount();
+        testContext.put(KEY, connection);
+    }
+
+    private class BrowserStackLocalConnection
+    {
+        private static final String PAC_FORMAT = "function FindProxyForURL(url, host) "
+                + "{ if (shExpMatch(url, \"%s\") || shExpMatch(host, \"*.browserstack.com\")) { return \"DIRECT\"; }"
+                + "return \"PROXY %s\"; }";
+
+        private final String localIdentifier;
+        private final AtomicInteger sessionCount;
+        private final Map<String, String> localParameters;
+        private final Local local;
+
+        BrowserStackLocalConnection(BrowserStackLocalOptions options) throws IOException
+        {
+            this.localIdentifier = UUID.randomUUID().toString();
+            this.sessionCount = new AtomicInteger(0);
+
+            Map<String, String> parameters = new HashMap<>();
+            parameters.put("localIdentifier", localIdentifier);
+            parameters.put("forcelocal", "true");
+            parameters.put("key", sauceLabsAccessKey);
+
+            if (options.getProxy() != null)
+            {
+                String pac = ResourceUtils.createTempFile("pac-" + localIdentifier, ".js",
+                        String.format(PAC_FORMAT, options.getSkipUrlsPattern(), options.getProxy())).toString();
+                parameters.put("-pac-file", pac);
+            }
+
+            this.localParameters = parameters;
+            this.local = new Local();
+        }
+
+        String getLocalIdentifier()
+        {
+            return localIdentifier;
+        }
+
+        void startConnection() throws Exception
+        {
+            LOGGER.atInfo().addArgument(localIdentifier)
+                           .log("Starting BrowserStack Local connection with {} identifier");
+            local.start(localParameters);
```

Comment:
why not to wait for start ending here?

Reply:
it will not prevent other threads from using this connection

Reply:
but still this connection won't be operable, and extra resources will be required to synchronize start completion

Reply:
start method already contains wait logic within it